services/mozilla-observatory/mozilla-observatory.service.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const label = 'observatory'
+const { BaseJsonService } = require('..')
+
+const Joi = require('joi')
+const schema = Joi.object({
+  grade: Joi.string()
+    .regex(/^[abcdef]$/i)
```

Comment:
I think this will need to support the optional `+` and `-` characters to account for plus/minus grades

Reply:
I've added them before, but removed them because `+` and `-` and not required by the badge code. I'm more in a minimal code approach where I only validate what I need. The only things the code need is the grade to starts with `abcdef`.. I don't care of the case. I don't care what follows (I'm not validating the Observatory API, but the input for my code).

Does that make sense to you?

Reply:
The schema validation is used directly on the API response, so it needs to handle whatever raw values we'll see in the API response. Right now, a request for any domain that has a +/- grade will result in schema validation failure and an error message of `invalid response data`

Reply:
So with `$` I guess. Do we really want to make the badge fail if one day Observatory adds extra characters to the grade? Willingly or by mistake (e.g. an extra space)?

Reply:
If we want to be strict, we also need to be case sensitive, and to make sur the score is a positive integer between 0 and 200 (actually it's 135 but it may changes over time)

Reply:
I can do that

Reply:
I can do that if you want:

```js
const schema = Joi.object({
  grade: Joi.string()
    .regex(/^[ABCDEF][+-]?$/)
    .required(),
  score: Joi.number()
    .integer()
    .min(0)
    .max(200) // At the time this was written max was 135, but may increase
    .required(),
}).required()
```

But really I'm not fan of "useless" code

Reply:
We're a small group of maintainers, working on a volunteer basis to support Shields.io and the community. There's a **ton** of services that Shields integrates with that we have to support and maintain, even though we don't personally use many of those services.

Because of that large scope of services, it's important that service classes are constructed in such a way that any maintainer can quickly grok, and support/debug without having to be intimately familiar with the underlying service.

The schema validation is one of the things that really helps us with that, because it serves as a form of documentation that quickly conveys what the API response structure is like without a maintainer having to dive deep into external documentation. As a design goal, we do also prefer stricter schema validation so that we can fail loudly and quickly, and then react accordingly. 

In this particular instance, it's beneficial to note in the schema validation that the API response includes letter grades, with an optional plus or minus value, to accurately describe and convey the nature of the grade values we can see from the API. 

Thanks for updating the grade regex! I'll give this PR final look over later today but I think it's just about ready to be merged.